src/layouts/docs/server.ts

```diff
@@ -1,108 +1,119 @@
+  ) => new RemoteContentLoader({ ...loaderOptions, ...extraOptions })
 
-  const { navData, mdxSource, githubFileUrl } = await _generateStaticProps({
-    basePath,
-    localContentDir: temporary_noop,
-    navDataFile: temporary_noop,
-    params,
-    product,
-    remarkPlugins: [[anchorLinks, { headings }]],
-  })
+  return {
+    getStaticPaths: async () => {
+      const paths = await getLoader().loadStaticPaths()
 
-  /**
-   * TODO: these will be different by product,
-   * can abstract these further later.
-   * Placing here because we need these links
-   * in the sidebar on all /waypoint/docs views.
-   */
-  const fullNavData = [
-    ...navData,
-    { divider: true },
-    {
-      title: 'HashiCorp Learn',
-      href: 'https://learn.hashicorp.com/waypoint',
-    },
-    {
-      title: 'Community Forum',
-      href: 'https://discuss.hashicorp.com/c/waypoint/51',
+      return {
+        fallback: 'blocking',
+        paths,
+      }
     },
-    {
-      title: 'Support',
-      href: 'https://support.hashicorp.com/',
-    },
-  ]
+    getStaticProps: async (ctx) => {
+      const headings = []
 
-  // Add fullPaths and ids to navData
-  const navDataWithFullPaths = prepareNavDataForClient(fullNavData, [
-    product.slug,
-    basePath,
-  ])
+      const loader = getLoader({ remarkPlugins: [[anchorLinks, { headings }]] })
 
-  const breadcrumbLinks = getDocsBreadcrumbs({
-    productPath: product.slug,
-    productName: product.name,
-    basePath,
-    baseName: 'Docs',
-    pathParts: params.page || [],
-    navData: navDataWithFullPaths,
-  })
+      const {
+        navData,
+        mdxSource,
+        githubFileUrl,
+      } = await loader.loadStaticProps(ctx)
+
+      /**
+       * TODO: these will be different by product,
+       * can abstract these further later.
+       * Placing here because we need these links
+       * in the sidebar on all /waypoint/docs views.
+       */
+      const fullNavData = [
+        ...navData,
+        { divider: true },
+        {
+          title: 'HashiCorp Learn',
+          href: 'https://learn.hashicorp.com/waypoint',
+        },
+        {
+          title: 'Community Forum',
+          href: 'https://discuss.hashicorp.com/c/waypoint/51',
+        },
+        {
+          title: 'Support',
+          href: 'https://support.hashicorp.com/',
+        },
+      ]
+
+      // Add fullPaths and ids to navData
+      const navDataWithFullPaths = prepareNavDataForClient(fullNavData, [
+        product.slug,
+        basePath,
+      ])
 
-  const finalProps = {
-    mdxSource,
-    layoutProps: {
-      headings,
-      navData: navDataWithFullPaths,
-      breadcrumbLinks,
-      githubFileUrl,
-      backToLink: {
-        text: `Back to ${product.name}`,
-        url: `/${product.slug}`,
-      },
+      const breadcrumbLinks = getDocsBreadcrumbs({
+        productPath: product.slug,
+        productName: product.name,
```

Comment:
It'd be nice to read this from the upcoming product context so we don't have to thread the product name/slug through here 💭

Reply:
I think that's tricky since these things are all happening server side and the Context can only be used client side, right?

Reply:
Correct, the way its currently implemented all the breadcrumb data is generated here. It would require a larger refactor in how breadcrumbs are rendered most likely

Reply:
Ah, okay! Wasn't sure if that's the kind of refactor you were thinking about. Right now my hot take is that if the breadcrumbs can all be generated ahead of time, then we doing so on the server side might be a valuable optimization? But I think it'd be good to weigh the pros and cons of each approach and I don't hold my current opinion very strongly. 😊

Reply:
Yeah, that's reasonable! I think there's probably a middle ground here somewhere. We have to iterate over and render the breadcrumb data even with it generated on the server, so we could conceivably access the context at the point and change the URL, maybe swapping out a placeholder or something.

```ts
{ title: '{productName}', url: '/{productPath}' }

// ...

const product = useCurrentProduct()

<a href={url.replace('{productPath}', product.slug)}>{title.replace('{productName}', product.name)}</a>
```

I'm a big fan of trying to avoid passing down arguments/props multiple levels if possible, but there are certainly trade-offs to both approaches.